examples/offline_inference/torchrun_example.py

```diff
@@ -0,0 +1,68 @@
+# see https://github.com/vllm-project/vllm/issues/11400 for
+# the motivation and use case for this example.
+
+import torch.distributed as dist
+
+from vllm import LLM, SamplingParams
+from vllm.distributed.parallel_state import get_world_group
+
+dist.init_process_group(backend="nccl")
+
+torch_rank = dist.get_rank()
+
+# Create prompts
+prompts = [
+    "Hello, my name is",
+    "The president of the United States is",
+    "The capital of France is",
+    "The future of AI is",
+]
+
+sampling_params = SamplingParams(temperature=0.8, top_p=0.95)
+
+# important: use `distributed_executor_backend="external_launcher"` so that
+# this llm engine/instance only creates one worker.
+# set different `gpu_memory_utilization` and `swap_space` for different ranks,
+# to test if all ranks agree on the same kv cache configuration.
+llm = LLM(model="facebook/opt-125m",
+          tensor_parallel_size=2,
+          distributed_executor_backend="external_launcher",
+          gpu_memory_utilization=0.9 if torch_rank == 0 else 0.7,
+          swap_space=3 if torch_rank == 0 else 4)
```

Comment:
I would against using example code to test functionality. It's not straightforward and could confuse users when they run the example and see the log. To test if all ranks agree on the same kv cache configuration, you should put it in unit tests.

Reply:
thanks for the feedback! they are separated now, it does look better!